README.md

```diff
@@ -10,4 +10,449 @@ TODO: description
+
+- A success message
+- A object with the created medication
+
+**Throws**
+
+- `401` if the user is not logged in
+- `400` If the name, generic_name, or dose is empty or a stream of empty spaces
+
+#### `PATCH /api/medications/:medicationId?` - Edit the information of a medication
+
+**Body**
+
+- `name` _{string}_ - The medication's name (optional)
+- `generic_name` _{string}_ - The medication's generic compound name (optional)
+- `dose` _{string}_ - The medication's dose (optional)
+- `notes` _{string}_ - Notes about the medication (optional)
+
+**Returns**
+
+- A success message
+- A object with the edited medication
+
+**Throws**
+
+- `401` if the user is not logged in
+- `403` if the user is not the owner of the medication
+- `400` If the name, generic_name, or dose is empty or a stream of empty spaces
+- `404` if the medicationId is invalid
+
+#### `DELETE /api/medications/:medicationId?` - Deactivates a medication
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `401` if the user is not logged in
+- `403` if the user is not the owner of the medication
+- `404` if the medicationId is invalid
+- `409` if the medicationId is already deactivated 
+
+### Entry
+
+#### `GET /api/entries` - Get all entries of the user
+
+**Returns**
+- The array of entries by the user sorted in descending order by date
+
+**Throws** 
+- `401` if the user is not logged in
+
+#### `POST /api/entries` - Make a new entry in the log
+
+**Body**
+- `type` _{string}_ - The type of the log entry
+- `detail` _{string}_ - The detail of the log entry
+- `notes` _{string}_ - The log entry's notes (optional)
+- `date` _{Date}_ - The date associated with the log entry (optional)
+
+**Returns**
+- A success message
+- The newly created entry object
+
+**Throws** 
+- `401` if the user is not logged in
+- `400` if the `type` or `detail` is empty or a stream of empty spaces
+
+#### `PATCH /api/entries/:entryId` - Editing an entry in the log
+
+**Body**
+- `type` _{string}_ - The type of the log entry (optional)
+- `detail` _{string}_ - The detail of the log entry (optional)
+- `notes` _{string}_ - The log entry's notes (optional)
+- `date` _{Date}_ - The date associated with the log entry (optional)
+
+**Returns**
+- A success message
+- The updated entry object
+
+**Throws** 
+- `401` if the user is not logged in
+- `404` if the `entryId` is invalid
+- `403` if the user is not the owner of the entry
+
+#### `DELETE /api/entries/:entryId` - Delete an entry in the log
+
+**Returns**
+- A success message
+
+**Throws** 
+- `401` if the user is not logged in
+- `404` if the `entryId` is invalid
+- `403` if the user is not the owner of the entry
+
+
+### Trend
+
+#### `GET /api/trends` - Get all trends for a given user  
+**Returns**  
+- An array of trends created by user with given username (taken from session)  
+**Throws**  
+- `401` if `user` is not found in session  
+
+#### `GET /api/trends/:itemId` - Get set of points that reference a given item (data entry)  
```

Comment:
What would we use this route for?

Reply:
yeah, this is the one that I was confused about. I was trying to make a 1:1 mapping of the methods defines in our concepts, and I think I just misunderstood this one. I'll just delete this one.

Reply:
because like @wli-linda said in a earlier thread, it's nice to have a 1:1 mapping of points to entries, so it doesn't make sense to have an entry with more than one point